lib/ansible/module_utils/basic.py

```diff
@@ -1450,6 +1473,9 @@ def set_fs_attributes_if_different(self, file_args, changed, diff=None, expand=T
         )
         return changed
 
+    def check_file_absent_if_check_mode(self, file_path):
+        return not os.path.exists(file_path) and self.check_mode
```

Comment:
Tiny optimization: check self.check_mode before checking os.path.exists.  That way in the common case we don't have to use a system call to determine if the path exists.

Reply:
Updated, thanks.